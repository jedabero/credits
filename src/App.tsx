import { useCallback } from 'react';
import './App.css';
import { addCard, removeCard, selectAllCreditCards } from './reducers/creditCards';
import { useAppDispatch, useAppSelector } from './hooks';

function App() {
  const dispatch = useAppDispatch();
  const handleAdd = useCallback(() => dispatch(addCard({
    id: 1,
    name: 'test',
    quota: 24633,
    balance: 0,
    cutoffDay: 15,
    cardNumber: '***465',
  })), [dispatch]);
  const cards = useAppSelector(selectAllCreditCards);
  const handleDelete = useCallback(() => dispatch(removeCard(1)), [dispatch]);

  return (
    <div className="App">
      <main>
        <h6>Tasks</h6>
        <p>Definir Credito</p>
        <p>Definir Movimiento</p>
        <p>Ver valores detallados del mes y de cada movimiento</p>
        <p>{JSON.stringify(cards)}</p>
        <button type="button" onClick={handleAdd}>Add Credit Card</button>
        <button type="button" onClick={handleDelete}>Delete Credit Card</button>
      </main>
    </div>
  );
}

export default App;
