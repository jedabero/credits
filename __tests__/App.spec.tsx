/* eslint-disable import/no-extraneous-dependencies */
import React, { ReactElement } from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import App from '../src/App';

describe('App test', () => {
  const renderer = createRenderer();
  let output: ReactElement;
  test('render App', () => {
    renderer.render(<App />);
    output = renderer.getRenderOutput<ReactElement>();
    expect(output).toMatchSnapshot();
  });
  test('should display movements', () => {
    expect(output.props.children[2]).toBeNull();
    output.props.children[1].props.onViewCardMovements('SOME ID');
    output = renderer.getRenderOutput<ReactElement>();
    expect(output.props.children[2]).not.toBeNull();
    expect(output).toMatchSnapshot();
  });
});
