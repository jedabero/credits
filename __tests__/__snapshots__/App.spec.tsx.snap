// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test render App 1`] = `
<div
  className="App"
>
  <main>
    <h6>
      Tasks
    </h6>
    <p>
      Definir Credito
    </p>
    <p>
      Definir Movimiento
    </p>
    <p>
      Ver valores detallados del mes y de cada movimiento
    </p>
  </main>
  <Cards
    onViewCardMovements={[Function]}
  />
</div>
`;

exports[`App test should display movements 1`] = `
<div
  className="App"
>
  <main>
    <h6>
      Tasks
    </h6>
    <p>
      Definir Credito
    </p>
    <p>
      Definir Movimiento
    </p>
    <p>
      Ver valores detallados del mes y de cada movimiento
    </p>
  </main>
  <Cards
    onViewCardMovements={[Function]}
  />
  <Movements
    cardId="SOME ID"
    onViewMovementHistory={[Function]}
  />
</div>
`;
